config/src/main/java/com/typesafe/config/ConfigMemorySize.java

```diff
@@ -0,0 +1,49 @@
+/**
+ *   Copyright (C) 2015 Typesafe Inc. <http://typesafe.com>
+ */
+package com.typesafe.config;
+
+/**
+ * An immutable class representing an amount of memory.  Use
+ * static factory methods such as {@link
+ * ConfigMemorySize#ofBytes(long)} to create instances.
+ */
+public final class ConfigMemorySize {
+    private final long bytes;
+
+    private ConfigMemorySize(long bytes) {
+        if (bytes < 0)
+            throw new IllegalArgumentException("Attempt to construct ConfigMemorySize with negative number: " + bytes);
+        this.bytes = bytes;
+    }
+
+    /** Constructs a ConfigMemorySize representing the given
+     * number of bytes.
+     */
+    public static ConfigMemorySize ofBytes(long bytes) {
+        return new ConfigMemorySize(bytes);
+    }
+
+    /** Gets the size in bytes.
+     */
+    public long toBytes() {
```

Comment:
good point though that we should have a toString. feeling bad about writing the "human friendly memory and/or duration string" function for the millionth time, I think I've written it a dozen times in my life, and I'm not sure a million times is an exaggeration for developers in general...  :-) maybe I'll go with `"ConfigMemorySize(" + bytes + ")"`